src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
                         .ContinueWith(x =>
                         {
                             return new ManagementCommandAck(x.Result.All(y => y));
-                        },
-                            TaskContinuationOptions.ExecuteSynchronously)
+                        }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
                         .PipeTo(sender);
-                })
-                .With<Quarantine>(quarantine =>
-                {
+                });
+
+            Receive<Quarantine>(quarantine =>
+            {
                     //Stop writers
-                    if (endpoints.WritableEndpointWithPolicyFor(quarantine.RemoteAddress) is Pass)
-                    {
-                        var pass = (Pass) endpoints.WritableEndpointWithPolicyFor(quarantine.RemoteAddress);
+                    var policy =
+                    Tuple.Create(_endpoints.WritableEndpointWithPolicyFor(quarantine.RemoteAddress), quarantine.Uid);
+                if (policy.Item1 is Pass && policy.Item2 == null)
+                {
+                    var endpoint = policy.Item1.AsInstanceOf<Pass>().Endpoint;
+                    Context.Stop(endpoint);
+                    _log.Warning("Association to [{0}] with unknown UID is reported as quarantined, but " +
+                    "address cannot be quarantined without knowing the UID, gating instead for {1} ms.", quarantine.RemoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds);
+                    _endpoints.MarkAsFailed(endpoint, Deadline.Now + _settings.RetryGateClosedFor);
+                }
+                else if (policy.Item1 is Pass && policy.Item2 != null)
```

Comment:
If the current write policy for this endpoint is `Pass` (writes are allowed to flow through to the transport) and we **do  have a UID** available for the `Address` we are going to `Quarantine`:

* stop the endpoint actor if the UID of the `Pass` policy matches that of the `Quarantine` one (means both applies to the same incarnation of the `ActorSystem` at the remote address.)
* If the `Quarantine` policy applies to a different UID, then ignore it.